### DISCUSSION TITLE
GitHub app to be opensourced?

### DISCUSSION THREAD

**Discussion by Krrishdhaneja** [community/community#22160] [New to GitHub]
<p>Um… I was having a doubt if GitHub app would ever be open source? Any ideas?</p>

**Comment by logankilpatrick** [comment, marked as answer]
<p>This is a great question, while I don’t work at GitHub nor have the answer, I would love to share my thoughts:</p>
<ul>
<li>GitHub recently open sourced their docs. See here: <a href="https://youtu.be/W49UEa4Y6sM" rel="noopener nofollow ugc">https://youtu.be/W49UEa4Y6sM</a> my impression is that this involved a tremendous amount of work for them internally.</li>
<li>Given how GitHub ~may~ work on the backend, making it open source would sort of require someone to actually run things on their end which given the way things are built, may not be possible without some sort of server side access.</li>
<li>You can support open source, without making everything open source. I personally have a number of repos that I haven’t made public for a variety of reasons but still consider myself a huge OSS fan.</li>
</ul>
<p>I’m sure there’s a whole host of other reasons why this hasn’t happened yet but hopefully we can continue to see small verticals open sourced where it makes sense.</p>
